ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java

```diff
@@ -210,6 +217,7 @@ private QueryPublisher createPullQueryPublisher(
             PullQueryResult::getPlanType).orElse(PullPhysicalPlanType.UNKNOWN);
         final RoutingNodeType routingNodeType = Optional.ofNullable(r).map(
             PullQueryResult::getRoutingNodeType).orElse(RoutingNodeType.UNKNOWN);
+        pullBandRateLimiter.add(startTimeNanos / 1000000000, responseBytes);
```

Comment:
Declare the literal as a constant or use `TimeUnit.NANOS.toMillis()`

Reply:
I was originally slightly worried about piggybacking on the metrics because I didn't want to throw an exception and break our metrics reporting, but the way we've implemented this, it's actually guaranteed to add the response, so I think this works well.